--- opera-31.0.1889.174/opera.desktop	2015-09-06 10:00:09.174694303 +0300
+++ opera-31.0.1889.174/opera.desktop	2015-09-06 10:01:59.496777742 +0300
@@ -1,15 +1,19 @@
 [Desktop Entry]
 Version=1.0
 Name=Opera
+Name[eo]=Opero
+Name[zu]=I Opera
 GenericName=Web browser
+GenericName[pl]=Przeglądarka WWW
 Comment=Fast and secure web browser
+Comment[pl]=Najszybsza przeglądarka WWW na świecie
 TryExec=opera
 Exec=opera %U
 Terminal=false
 Icon=opera
 Type=Application
 Categories=Network;WebBrowser;
-MimeType=text/html;text/xml;application/xhtml_xml;x-scheme-handler/http;x-scheme-handler/https;x-scheme-handler/ftp;
+MimeType=text/html;text/xml;application/xhtml+xml;x-scheme-handler/http;x-scheme-handler/https;x-scheme-handler/ftp;
 Actions=NewWindow;NewPrivateWindow;
 
 [Desktop Action NewWindow]
