--- opera-10.51-6252.i386.linux/share/applications/opera-browser.desktop~	2010-03-20 15:38:12.000000000 +0200
+++ opera-10.51-6252.i386.linux/share/applications/opera-browser.desktop	2010-03-20 15:40:34.551656601 +0200
@@ -2,6 +2,10 @@
 Type=Application
 Version=1.0
 Name=Opera
+Name[af]=opera
+Name[eo]=Opero
+Name[pl]=Opera
+Name[zu]=I Opera
 GenericName=Web browser
 GenericName[bs]=Web preglednik
 GenericName[de]=Web-Browser
@@ -16,6 +20,7 @@
 GenericName[nb]=Nettleser
 GenericName[nl]=webbrowser
 GenericName[nn]=Nettlesar
+GenericName[pl]=Przeglądarka WWW
 GenericName[pt]=Navegador Web
 GenericName[pt_BR]=Navegador
 GenericName[ro]=Navigator de web
@@ -30,6 +30,7 @@
 GenericName[xh]=Umkhangeli Zincwadi Zokubhaliweyo
 GenericName[zu]=Umkhangeli zincwadi we Web
 Comment=Fast and secure web browser and Internet suite
+Comment[pl]=Najszybsza przeglądarka WWW na świecie
 Icon=opera-browser
 TryExec=/usr/bin/opera
 Exec=/usr/bin/opera %U

