--- opera-9.02-20060919.5-shared-qt.i386-en-434/install.sh~	2006-12-01 18:43:27.800063859 +0200
+++ opera-9.02-20060919.5-shared-qt.i386-en-434/install.sh	2006-12-01 18:44:24.201331154 +0200
@@ -843,13 +843,19 @@
     wrapper_contain="#!/bin/sh
 
 # Required for Session Management
-case \$0 in /*) OPERA_SCRIPT_PATH=\$0;; *) OPERA_SCRIPT_PATH=`/bin/pwd`/\$0;; esac
+case \$0 in /*) OPERA_SCRIPT_PATH=\$0;; *) OPERA_SCRIPT_PATH=/usr/bin/\$0;; esac
 export OPERA_SCRIPT_PATH
 
 # Location of the Opera binaries
 OPERA_BINARYDIR=${str_localdirexec}
 export OPERA_BINARYDIR
 
+# set JAVA_HOME from jpackage-utils if available
+if [ -f /usr/share/java-utils/java-functions ]; then
+	. /usr/share/java-utils/java-functions
+	set_jvm
+fi
+
 # Parse commandline parameters
 toset=
 _QTSTYLE_FROM_CMDLINE_='NO'
