--- opera-10.51-6252.i386.linux/opera-next~	2010-03-20 02:45:01.000000000 +0200
+++ opera-10.51-6252.i386.linux/opera-next	2010-03-20 13:32:03.132427427 +0200
@@ -1,4 +1,9 @@
 #!/bin/sh
-export OPERA_DIR=${OPERA_DIR:-"${0%/*}"/share/opera-next}
-export OPERA_PERSONALDIR=${OPERA_PERSONALDIR:-"${0%/*}"/profile}
-exec "${0%/*}"/lib/opera-next/opera-next "$@"
+# set JAVA_HOME from jpackage-utils if available
+if [ -f /usr/share/java-utils/java-functions ]; then
+	. /usr/share/java-utils/java-functions
+	set_jvm
+fi
+export OPERA_DIR=/usr/share/opera-next
+export OPERA_PERSONALDIR=$HOME/.opera-next
+exec /usr/lib/opera-next/opera-next "$@"
